teamC/sungsik/app.js

```diff
@@ -31,10 +31,53 @@ app.use(express.json());
 app.use(cors());
 app.use(morgan("combined"));
 
+
+// connection check
 app.get('/ping', (req, res) => {
     res.status(200).json({ message:'pong' })
 })
 
+
+//create users
+app.post('/users', async (req, res, next) => {
+    const { name, email, profile_image, password } = req.body
+    
+    const check = await database.query(
+        `SELECT email FROM users WHERE users.email = ?`
+        ,email
+    )
+    
+    if (check.length === 0) {
+        await database.query(
+            `INSERT INTO users(
+                name,
+                email,
+                profile_image,
+                password
+            ) VALUES (?, ?, ?, ?);
+            `,
+            [ name, email, profile_image, password ]
+        );
+        res.status(200).json({ message:'userCreated'})
+    } else {
+        if (check[0].email === email) {
+            res.status(409).json({ message:'userOverlaped'})
+        } else {
+            await database.query(
+                `INSERT INTO users(
+                    name,
+                    email,
+                    profile_image,
+                    password
+                ) VALUES (?, ?, ?, ?);
+                `,
+                [ name, email, profile_image, password ]
+            );
+            res.status(200).json({ message:'userCreated'})
+        }
+    }
+})
```

Comment:
아래와 같이 수정하면 더 깔끔할 것 같네요~
```suggestion
    const user = await database.query(`
        SELECT 
            email 
        FROM users 
        WHERE users.email = ?` , [email]
    )
    
    if (!user.length === 0) { res.status(409).json({ message:'userOverlaped'})}
    
      await database.query(
          `INSERT INTO users(
              name,
              email,
              profile_image,
              password
          ) VALUES (?, ?, ?, ?);
          `,
          [ name, email, profile_image, password ]
      );
        res.status(200).json({ message:'userCreated'})
    }
})
```